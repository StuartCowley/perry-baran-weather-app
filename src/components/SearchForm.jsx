import React, { useRef } from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

function SearchForm({ handleSearch, placeholder }) {
  const searchRef = useRef(null);

  const handleSubmit = (e) => {
    e.preventDefault();
    handleSearch(searchRef.current.value);
    searchRef.current.value = "";
  };

  const handleKeyDown = (e) => e.code === "Enter" && handleSubmit(e);

  return (
    <div className="search-form">
      <input
        type="text"
        ref={searchRef}
        onKeyDown={handleKeyDown}
        placeholder={placeholder}
      />
      <button type="submit" onClick={handleSubmit}>
        Search
      </button>
    </div>
  );
}

SearchForm.defaultProps = {
  placeholder: "",
};

SearchForm.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
};

export default SearchForm;
