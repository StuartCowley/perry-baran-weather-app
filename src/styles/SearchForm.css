.search-form {
  height: 27px;
  width: calc(100% - 50px);
  max-width: 600px;
  margin: auto 0;
  border: var(--border-dark-blue);
  border-radius: 2px;
  background-color: var(--almost-white);
  overflow: hidden;
}

.search-form input {
  all: unset;
  box-sizing: border-box;
  height: 100%;
  width: calc(100% - 65px);
  padding: 5px;
  font-size: 1.2rem;
}

.search-form button {
  all: unset;
  box-sizing: border-box;
  height: 100%;
  width: 65px;
  border-left: var(--border-dark-blue);
  background-color: var(--medium-blue);
  color: var(--almost-white);
  font-size: 1rem;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  transition-duration: 200ms;
}

.search-form button:hover {
  background-color: var(--dark-blue);
}