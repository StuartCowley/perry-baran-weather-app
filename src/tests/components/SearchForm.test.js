import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const handleSearch = jest.fn();
  let screen;

  describe("including placeHolder", () => {
    const placeHolder = "placeholder";
    beforeEach(() => {
      screen = render(
        <SearchForm handleSearch={handleSearch} placeholder={placeHolder} />
      );
    });

    test("snapshot", () => {
      const { asFragment } = screen;

      expect(asFragment()).toMatchSnapshot();
    });

    test("renders correctly", () => {
      const { getByRole } = screen;
      const button = getByRole("button");
      const input = getByRole("textbox");

      expect(button).toHaveTextContent(/search/i);
      expect(button).toHaveAttribute("type", "submit");
      expect(input).toBeInTheDocument();
      expect(input.value).toBe("");
      expect(input).toHaveAttribute("placeholder", placeHolder);
    });

    test("inputs", async () => {
      const { getByRole } = screen;
      const button = getByRole("button");
      const input = getByRole("textbox");
      const string = "string";

      fireEvent.change(input, { target: { value: string } });
      expect(input.value).toBe(string);
      fireEvent.click(button);
      expect(handleSearch).toBeCalledWith(string);
      expect(input.value).toBe("");
    });

    test("can press enter in search input to search", () => {
      const { getByRole } = screen;
      const input = getByRole("textbox");
      const string = "string";

      fireEvent.change(input, { target: { value: string } });
      expect(input.value).toBe(string);
      fireEvent.keyDown(input, { code: "Enter", charCode: 13 });
      expect(handleSearch).toBeCalledWith(string);
      expect(input.value).toBe("");
    });
  });

  describe("exluding placeholder", () => {
    beforeEach(() => {
      screen = render(<SearchForm handleSearch={handleSearch} />);
    });

    test("renders correctly without placeholder", () => {
      const { getByRole } = screen;
      const input = getByRole("textbox");

      expect(input).toHaveAttribute("placeholder", "");
    });
  });
});
