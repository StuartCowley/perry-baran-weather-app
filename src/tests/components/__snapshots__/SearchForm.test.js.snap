// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm including placeHolder snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="placeholder"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
